test/hotspot/jtreg/runtime/logging/AsyncDeathTestDebug.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8349755
+ * @summary Perform recursive logging in async UL and observe a crash
+ * @requires vm.flagless
+ * @requires vm.debug
+ * @library /test/lib
+ * @modules java.base/jdk.internal.misc
+ *          java.management
+ * @run driver AsyncDeathTestDebug
+ */
+
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.process.OutputAnalyzer;
+
+public class AsyncDeathTestDebug {
+    public static void main(String[] args) throws Exception {
+        ProcessBuilder pb =
+            ProcessTools.createLimitedTestJavaProcessBuilder("-Xlog:async", "-Xlog:all=debug");
```

Comment:
If we are going to crash we should add `-XX:-CreateCoredumpOnCrash`

Reply:
Also does this mean we are going to crash on the very first log_debug? You have two logging statements that will trigger the crash but only one of them can be reached with the way the testing is structured.

Reply:
When a test knows it will crash we don't want to generate a coredump as we don't care about it. Dumping can take a significant time on some systems leading to timeouts.

Reply:
Thanks David, I missed the `-` and was under the pretension that we don't create coredumps by default :).